src/nvim/ex_docmd.c

```diff
@@ -6858,10 +6882,25 @@ void ex_cd(exarg_T *eap)
       new_dir = NameBuff;
     }
 #endif
-    if (new_dir == NULL || vim_chdir(new_dir))
+    if (new_dir == NULL || vim_chdir(new_dir)) {
```

Comment:
Good question. I was just taking the original code as it was. The only way for `new_dir` to be `NULL` is if `eap->arg` is `NULL`, which if I understand should never happen. Removed the check.